### ISSUE TITLE
Fresh install does not load

### ISSUE THREAD

**Issue by timnovinger** [causefx/Organizr#253]
After a fresh install (for the first time), I'm unable to see anything but the loading animation. My Nginx log shows the following...

```
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr: PHP Notice:  Undefined index: databaseLocation in /var/www/html/organizr/user.php on line 34
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr: PHP Notice:  Undefined index: databaseLocation in /var/www/html/organizr/user.php on line 35
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr: PHP Fatal error:  Uncaught PDOException: SQLSTATE[HY000] [14] unable to open database file in /var/www/html/organizr/user.php:237
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr: Stack trace:
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr: #0 /var/www/html/organizr/user.php(237): PDO->__construct('sqlite:/users.d...')
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr: #1 /var/www/html/organizr/index.php(72): User->__construct('registration_ca...')
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr: #2 {main}
2017-04-09 08:54:21: (mod_fastcgi.c.2520) FastCGI-stderr:   thrown in /var/www/html/organizr/user.php on line 237
```

I see that there is no databaseLocation.ini.php file in the organizer directory. The check.php file shows everything as all green.

Safari and Chrome are also both showing the following console errors if this helps any:

`[Error] ReferenceError: Can't find variable: addToHomescreen Global Code (organizr:466)`
`[Error] ReferenceError: Can't find variable: $ Global Code (organizr:1097)`

Here is a listing of the directory it is in:

```
➜  organizr git:(master) pwd
/var/www/html/organizr
➜  organizr git:(master) ll
total 612K
drwxrwsr-x 10 www-data www-data 4.0K Apr  9 08:55 .
drwxrwxr-x  6 www-data www-data 4.0K Apr  9 08:53 ..
-rwxr-xr-x  1 www-data www-data  116 Apr  9 08:38 ajax.php
-rwxr-xr-x  1 www-data www-data  135 Apr  9 08:38 ajax_remove_file.php
-rwxr-xr-x  1 www-data www-data 2.0K Apr  9 08:38 ajax_upload_file.php
-rwxr-xr-x  1 www-data www-data 1.2K Apr  9 08:38 auth.php
drwxr-xr-x 25 www-data www-data 4.0K Apr  9 08:38 bower_components
-rwxr-xr-x  1 www-data www-data 6.0K Apr  9 08:38 check.php
-rwxr-xr-x  1 www-data www-data  18K Apr  9 08:38 class.uploader.php
-rw-r--r--  1 www-data www-data  303 Apr  9 08:38 composer.json
-rw-r--r--  1 www-data www-data  30K Apr  9 08:38 composer.lock
drwxr-xr-x  2 www-data www-data 4.0K Apr  9 08:38 config
-rw-r--r--  1 www-data www-data   25 Apr  9 08:38 _config.yml
drwxr-xr-x  4 www-data www-data 4.0K Apr  9 08:38 css
-rwxr-xr-x  1 www-data www-data 5.1K Apr  9 08:38 downloader.php
-rwxr-xr-x  1 www-data www-data 6.9K Apr  9 08:38 error.php
-rwxr-xr-x  1 www-data www-data  40K Apr  9 08:38 functions.php
drwxr-xr-x  8 www-data www-data 4.0K Apr  9 08:53 .git
-rw-r--r--  1 www-data www-data  443 Apr  9 08:38 .gitattributes
-rw-r--r--  1 www-data www-data  818 Apr  9 08:38 .gitignore
-rwxr-xr-x  1 www-data www-data  26K Apr  9 08:38 homepage.php
-rwxr-xr-x  1 www-data www-data 1.4K Apr  9 08:38 image.php
drwxr-xr-x  5 www-data www-data 4.0K Apr  9 08:38 images
-rwxr-xr-x  1 www-data www-data  78K Apr  9 08:38 index.php
drwxr-xr-x  6 www-data www-data 4.0K Apr  9 08:38 js
drwxr-xr-x  2 www-data www-data 4.0K Apr  9 08:38 lang
-rw-r--r--  1 www-data www-data  35K Apr  9 08:38 LICENSE
-rw-r--r--  1 www-data www-data 4.1K Apr  9 08:38 README.md
-rwxr-xr-x  1 www-data www-data  585 Apr  9 08:38 register.php
-rw-r--r--  1 www-data www-data   25 Apr  9 08:38 robots.txt
-rwxr-xr-x  1 www-data www-data 196K Apr  9 08:38 settings.php
-rwxr-xr-x  1 www-data www-data  447 Apr  9 08:38 submitCSS.php
-rwxr-xr-x  1 www-data www-data   19 Apr  9 08:38 translate.php
-rwxr-xr-x  1 www-data www-data 4.3K Apr  9 08:38 updatedb.php
-rwxr-xr-x  1 www-data www-data  48K Apr  9 08:38 user.php
drwxr-xr-x 10 www-data www-data 4.0K Apr  9 08:38 vendor
```

How can I get this working?

**Comment by causefx** [comment]
are you running this on a pi?

**Comment by wbernier8** [comment]
I have the same problem. Fresh install on a Raspberry Pi 3 with Apache and nothing but the loading animation

**Comment by timnovinger** [comment]
@causefx nope. This is running on an Ubuntu Server 16.04 LTS install.

**Comment by causefx** [comment]
That's odd. Did the very first page open when u installed it? Asking for database path? Also do u have a dns blocker or something like pihole?

**Comment by timnovinger** [comment]
I installed by cloning the git repo. The very first (and only) thing I've seen has been that loader and that's it. Aside from looking at check.php of course.

I DO have pi-hole running on this box/network now that you mention it. Let me look at those logs...

**Comment by wbernier8** [comment]
I also have Pi-hole. I'll look through my logs too. @timnovinger please let me know if you find anything

**Comment by timnovinger** [comment]
@wbernier8 I'm not seeing anything yet. Looks like Organizr is pulling assets from the local machine so I'm not sure how it would even matter? It doesn't appear to be making external requests.

**Comment by timnovinger** [comment]
I disabled pi-hole and it made zero difference @causefx, @wbernier8.

**Comment by wbernier8** [comment]
As you said, @causefx, I think it's an issue with the database path. I keep getting this error: 
`Warning: parse_ini_file(databaseLocation.ini.php): failed to open stream: No such file or directory in /var/www/html/Organizr/user.php on line 33`
I haven't been able to make it to the first page to be able to specify the database path so I imagine that's why this error keeps happening.

**Comment by timnovinger** [comment]
Those *.ini.php files simply don't exist. I assume there's an installer wizard that generates those once you get to the first page? But if that is the case, we just aren't making it that far along.

Throwing in an empty `databaseLocation.ini.php` file just results in other errors as one would expect.

**Comment by causefx** [comment]
The file that calls that ini file shouldn't be included until setup has completed. Can you load the page and then view page source to see if it lists any php errors.

**Comment by causefx** [comment]
Also what version of php are you running

**Comment by timnovinger** [comment]
```
PHP 7.0.17-3+deb.sury.org~yakkety+1 (cli) (built: Apr  8 2017 08:46:21) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.17-3+deb.sury.org~yakkety+1, Copyright (c) 1999-2017, by Zend Technologies
```

I'm not seeing any PHP errors in the markup. I'm seeing what we've pasted in already in earlier comments in my Nginx log though.

I am seeing JS errors however...

![screen shot 2017-04-09 at 21 26 25](https://cloud.githubusercontent.com/assets/28424/24842905/597a531c-1d6b-11e7-9ca9-0dd9439946e6.png)
![screen shot 2017-04-09 at 21 26 42](https://cloud.githubusercontent.com/assets/28424/24842903/59750862-1d6b-11e7-8d64-ebb77e573f12.png)
![screen shot 2017-04-09 at 21 26 49](https://cloud.githubusercontent.com/assets/28424/24842904/5975261c-1d6b-11e7-9ccf-61057b0cfab4.png)

**Comment by wbernier8** [comment]
```
PHP 5.6.29-0+deb8u1 (cli) (built: Dec 17 2016 06:04:43) 
Copyright (c) 1997-2016 The PHP Group
Zend Engine v2.6.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
```

I have the same two errors as @timnovinger, line 466 and line 1097

<img width="568" alt="organizr errors" src="https://cloud.githubusercontent.com/assets/9525225/24843020/a44dfeb0-1d6c-11e7-88a6-e4a774d528cc.png">

**Comment by causefx** [comment]
can you delete the script that adds addtohomescreen on index as well as they line that calls it?

`<script src="js/addtohomescreen.js"></script>`

`<script>addToHomescreen();</script>`

**Comment by Cerothen** [comment]
@timnovinger 

Can you confirm that the www-data user actually has permission to the folder that organizr is in? I can see that all the folders and files in the directory are owned by www-data.

If the above doesnt work can you make a file called `databaseLocation.ini.php` and fill it with the following:
`
; <?php die("Access denied"); ?>
databaseLocation = "./config"
timezone = "Africa/Abidjan"
`

Once this is done browse the index.php file again in your browser.

**Comment by timnovinger** [comment]
Here's the directory and permission structure:

```
➜  html ll /var
drwxr-xr-x  3 root root   4.0K Apr  9 09:30 www

➜  html ll /var/www
drwxrwxr-x  6 www-data www-data 4.0K Apr  9 15:46 html

➜  html ll /var/www/html
drwxrwsr-x 10 www-data www-data 4.0K Apr  9 17:49 organizr
```

And the PHP (in browser) error that is thrown once I manually add that line of code you sent:
```
Notice: Undefined index: databaseLocation in /var/www/html/organizr/user.php on line 34

Notice: Undefined index: databaseLocation in /var/www/html/organizr/user.php on line 35

Fatal error: Uncaught PDOException: SQLSTATE[HY000] [14] unable to open database file in /var/www/html/organizr/user.php:237 Stack trace: #0 /var/www/html/organizr/user.php(237): PDO->__construct('sqlite:/users.d...') #1 /var/www/html/organizr/index.php(72): User->__construct('registration_ca...') #2 {main} thrown in /var/www/html/organizr/user.php on line 237
```

And the error that is thrown in `/var/log/lighttpd/error.log`:
```
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr: PHP Notice:  Undefined index: databaseLocation in /var/www/html/organizr/user.php on line 34
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr: PHP Notice:  Undefined index: databaseLocation in /var/www/html/organizr/user.php on line 35
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr: PHP Fatal error:  Uncaught PDOException: SQLSTATE[HY000] [14] unable to open database file in /var/www/html/organizr/user.php:237
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr: Stack trace:
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr: #0 /var/www/html/organizr/user.php(237): PDO->__construct('sqlite:/users.d...')
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr: #1 /var/www/html/organizr/index.php(72): User->__construct('registration_ca...')
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr: #2 {main}
2017-04-09 17:55:23: (mod_fastcgi.c.2520) FastCGI-stderr:   thrown in /var/www/html/organizr/user.php on line 237
```

**Comment by causefx** [comment]
@timnovinger new version is out

can you try that?

**Comment by timnovinger** [comment]
@causefx 

I'm seeing these errors now:
```
Notice: date_default_timezone_set(): Timezone ID '' is invalid in /var/www/html/organizr/user.php on line 26

Fatal error: Uncaught PDOException: SQLSTATE[HY000] [14] unable to open database file in /var/www/html/organizr/user.php:160 Stack trace: #0 /var/www/html/organizr/user.php(160): PDO->__construct('sqlite:/users.d...') #1 /var/www/html/organizr/index.php(67): User->__construct('registration_ca...') #2 {main} thrown in /var/www/html/organizr/user.php on line 160
```

**Comment by Cerothen** [comment]
@timnovinger 
Can you post the contents of your ./config/config.php?

Looking at this `PDO->__construct('sqlite:/users.d...')`

I would guess that in your config.php you will have a line that looks like this:
`"database_Location" => "",` while it should be something like `"database_Location" => "./config/",`

**Comment by timnovinger** [comment]
```
<?php
return array(
        "database_Location" => "/",
        "user_home" => "users/",
        "embyURL" => "",
        "plexURL" => "",
        "nzbgetURL" => "",
        "sabnzbdURL" => "",
        "headphonesURL" => ""
);
```

> At the first setup screen what are you putting in?

Nothing. I never get so far as to see a form of any kind. I haven't input anything in...it's stuck on the literal first load.

**Comment by Cerothen** [comment]
strange, there is a setup form that should come up.

can you change the lines to match below

```
<?php
return array(
	"database_Location" => "./config/",
	"user_home" => "./config/users/",
);

**Comment by timnovinger** [comment]
That results in a new error.

```
2017-04-13 06:18:10: (mod_fastcgi.c.2520) FastCGI-stderr: PHP Notice:  date_default_timezone_set(): Timezone ID '' is invalid in /var/www/html/organizr/user.php on line 26
```

**Comment by Cerothen** [comment]
`"timezone" => "America/Toronto",`

**Comment by timnovinger** [comment]
Well.....no more PHP errors but still stuck looking at the loading animation and no setup form or UI of any kind. The console is still complaining about those JS errors from earlier.

**Comment by Cerothen** [comment]
Hmm, I will revisit the install process to hopefully avoid your weird config file issues. As far as the JavaScript errors, I will also have to look but just as an off question what browser is it? Chome?

**Comment by timnovinger** [comment]
Thanks @Cerothen. And I've seen the errors consistently in Safari, Chrome, and Firefox. I could test in IE for you but that would require firing up some VMs. However, I'm sure it's the same error in there as well.

**Comment by Cerothen** [comment]
No, its fine. Just was curious.

**Comment by timnovinger** [comment]
Bump for any more help

**Comment by wbernier8** [comment]
Tried another fresh install and still stuck at the loading animation. 
Organizr/check.php is showing all green. Same two errors as before on lines 466 and 1097.

**Comment by Cerothen** [comment]
Can you guys try using the latest?

https://github.com/causefx/Organizr/tree/cero-dev

Also when point out line item issues can you please also outline the file please, thanks.

**Comment by timnovinger** [comment]
@Cerothen @causefx I apologize....this ended up **not** being an issue caused by Organizr but instead by Pihole, although not in the way I thought.

TL;DR — The solution was to run Pihole in a sandboxed VM.

So, even after disabling Pihole, the problem (JS not loading fully so I never saw the initial screen) persisted, so I didn't think Pihole was the cause of the issue....

The cause ended up being a result of running Organizr via Lighttp in `/var/www/html/organizr` _WHILE_ Pihole was also installed in `/var/www/html/pihole`. Pihole had set up Lighttp as part of its install and had put in place some really restrictive rules that prevented anything from loading outside of `WEBROOT/admin`.

So Lighttp was blocking asset requests before Pihole even saw them — which is why disabling it had no effect. I should have noticed this earlier because Muximux wasn't working either (which prompted me to try out Organizr).

**Comment by SubNoizey** [comment]
@timnovinger Do you mind giving a quick little how to? I'm currently experiencing the same issue

**Comment by TheFirstQuestion** [comment]
In case anyone runs across this issue and needs a quick fix...
The issue does indeed lie with installing both Pihole and Organizr. I chose to uninstall Pihole using `pihole uninstall` and everything is now working fine.
